exercise/submission_models.py

```diff
@@ -290,6 +303,10 @@ def set_error(self):
     def is_graded(self):
         return self.status in (self.STATUS.READY, self.STATUS.UNOFFICIAL)
 
+    @property
+    def lang(self):
+        return self.meta_data.get('lang', None)
```

Comment:
What if the `meta_data` is empty? Is it a dictionary then or does it become `None` or an empty string? In other words, does the `get` method always exist? You have modified the `Submission.objects.create_from_post` method, but sometimes new submissions are created without using that method. Additionally, when this update is deployed to production, all old existing submissions have empty meta data.